treatment-service/models/storage_test.go

```diff
@@ -523,6 +523,9 @@ func TestDumpExperiments(t *testing.T) {
 		time.Date(2022, 1, 2, 3, 5, 7, 0, time.UTC),
 	), segmentersType)
 	require.NoError(t, err)
+
+	// Manually set experiment id so it matches the id in the dump
+	e.Id = 81
```

Comment:
This id is now set manually because for some reason the unit tests are failing as this ID field (it's generated randomly [here](https://github.com/caraml-dev/xp/blob/6ef47af4e979f7033133c40f3eff281a7ae85114/treatment-service/models/storage_test.go#L54)) is now being compared with a hardcoded id value found [here](https://github.com/caraml-dev/xp/blob/6ef47af4e979f7033133c40f3eff281a7ae85114/treatment-service/testdata/experiments_dump.json#L20). This might be happening because of the upgrade to the proto version I believe.

Reply:
How was this even working previously when the id is generated randomly but the expected id is set to 81 ?

Reply:
I have no idea to be honest 😂 There's actually another field in the same struct (`name`) that's being generated with a new UUID each time https://github.com/caraml-dev/xp/blob/6ef47af4e979f7033133c40f3eff281a7ae85114/treatment-service/models/storage_test.go#L47 and somehow this doesn't fail the deepequal comparison. I don't think we changed anything else so it must've been one of the upgrades to the protobuf/reflect packages that changed something 🤔